pkg/testframework/framework.go

```diff
@@ -0,0 +1,59 @@
+package testframework
+
+import (
+	"time"
+
+	// "github.com/davecgh/go-spew/spew"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+var (
+	// AsyncOperationTimeout is how long we want to wait for asynchronous
+	// operations to complete. ForeverTestTimeout is not long enough to create
+	// several replicas and get them available on a slow machine.
+	// Setting this to 5 minutes:w
+
+	AsyncOperationTimeout = 5 * time.Minute
+)
+
+// DeleteCompletely sends a delete request and waits until the resource and
```

Comment:
Can we use this for any resource?  It appears so, and trust it will always delete all dependents.

Reply:
yes it should